docs/manual/developer/07_understanding_build_system.md

```diff
@@ -122,3 +122,49 @@ refer to their help text for more information and usage:
 
 Many of these utilities are simply front-ends over code in the SSG Python
 module located under `ssg/`.
+
+## How OVAL is Built
+
+The build of the OVAL document takes place in two steps.
+
+### 1. Combination of OVALs
+
+In the first step, all available and applicable for platform OVAL checks are builded into a single unlinked OVAL document stored in the `build/${PRODUCT}/oval-unlinked.xml` directory.
+The `oval-unlinked.xml` document is generated using the `combine_ovals.py` script.
+The OVAL shortcuts are loaded into the OVAL Document object in the order that the benchmark checks are loaded first, followed by the shared directory checks.
+If the shorthand is already loaded into the OVAL Document object, it is skipped.
+
+Steps of loading the OVAL shorthand:
+
+1. The OVAL Shorthand file is loaded as a string, and in the case of templated Shorthand, it is expanded using Jinja macros before loading.
+2. The OVAL Shorthand string is processed by the OVAL Document object.
+   1. The OVAL Shorthand string is loaded into the OVAL Shorthand object.
+   2. the OVAL Shorthand object is validated
+      It is checked:
+       - Whether the OVAL definitions are applicable to the product.
+       - If there is an OVAL definition in the shorthand with the same id as the given rule_id.
+3. If the OVAL Shorthand object is valid, it is added to the OVAL Document object.
+
+After all OVAL Shorthands are loaded, the affected platforms of the loaded OVAL definitions are completed.
+And then the OVAL document is saved as an XML file in `build/${PRODUCT}/oval-unlinked.xml`.
+
+### 2. Linking OVAL Document
+
+The second step is performed when building an XCCDF document using the `build_xccdf.py` script.
+In this step, the `oval-unlinked.xml` document from the previous step is linked to the XCCDF document being built.
```

Comment:
It would be great to explain that in this context the term "linking" means alignment of IDs between rules and checks.